enhancement(logplex source): annotate logs with query parameters

CGA1123 opened the pull request:
Allow annotation of logs via query parameters for http + logplex source,
in a similar fashion as is currently supported for HTTP headers.

This is particularly useful with the logplex source as it allows for a
single vector instance to be used as a drain for multiple Heroku
applications.

Closes #4732

Signed-off-by: Christian Gregg <christian@bissy.io>

CGA1123 commented on src/sources/http.rs:
```diff
@@ -124,6 +134,27 @@ fn add_headers(
     events
 }
 
+fn add_query_params(
```
This is duplicated exactly in `logplex`, maybe could be pulled into `util/http.rs`?

jszwedko replied:
:+1: I think that's a good idea. Maybe `src/sources/util/http.rs` since I think only sources would make use of it.

jszwedko commented on src/sources/logplex.rs:
```diff
@@ -23,6 +24,7 @@ use warp::http::{HeaderMap, StatusCode};
 #[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct LogplexConfig {
     address: SocketAddr,
+    query_params: Vec<String>,
```
Since this will be pulled out as the public facing user configuration, could we call it `query_parameters`?

jszwedko commented on src/sources/logplex.rs:
```diff
@@ -108,6 +121,27 @@ fn decode_message(body: Bytes, header_map: HeaderMap) -> Result<Vec<Event>, Erro
+        let value = query_params
+            .get(query_param_name)
+            .map(String::as_bytes)
+            .unwrap_or_default();
```
I'm seeing that this matches `add_headers`, but I'm wondering if we shouldn't map `None` to `event::Value::Null` rather than the empty string. That would help distinguish parameters that are missing from those that exist, but are empty. I think `add_headers` may have predated `event::Value::Null`; I wouldn't mind seeing that updated too.

jszwedko reviewed (commented):
Thanks for this @CGA1123 !

Just one note about mapping non-existent parameters, but I think the rest looks good.

We'll want to add this to the docs too. If you feel like taking a shot, you can find them in `docs/`, otherwise we can add them once this is merged.

CGA1123 commented:
Hi @jszwedko thanks for the review, I think I have addressed your comments!
I took a shot at updating the docs as well 😄 

CI seems to have hanged on the mac build though, not sure why...

jszwedko reviewed (approved):
Looks great! Thanks @CGA1123 . I kicked CI to run it again and will merge once it is green.

CGA1123 commented:
Yay! Thanks @jszwedko 😄 

Will be trying this out in the coming days on our apps at work I think 🚀
